### PR TITLE
Refactor controller - Closes #6707

### LINKED ISSUE #6707: Application constructor should not open handler
### Description

Framework constructor should not open any handler.
All handlers (file, DB or port etc) should be opened on `run`.
All open handles should be closed when shutdown.

### Acceptance Criteria

- No open handle remains after shutting down

### PR DESCRIPTION
**Pull request by shuse2**
### What was the problem?

This PR resolves #6707 

### How was it solved?

- Refactor controller to handle the plugins
- Update constructor not to use open handles (zeromq class opens handler, so moved to `start`)

### How was it tested?

- All unit test should pass
- test app should start (although it will fail on for ` forgersList is not iterable`

### CODE REVIEW

File: framework/src/application.ts [hunk 1]
```diff
@@ -243,20 +219,20 @@ export class Application {
...
-			await this._loadPlugins();
+			await this._controller.start();
 			await this._node.start();
-			this.logger.debug(this._controller.bus.getEvents(), 'Application listening to events');
-			this.logger.debug(this._controller.bus.getEndpoints(), 'Application ready for actions');
+			this.logger.debug(this._controller.getEvents(), 'Application listening to events');
+			this.logger.debug(this._controller.getEndpoints(), 'Application ready for actions');
 
-			this._channel.publish(APP_EVENT_READY);
+			this._controller.channel.publish(APP_EVENT_READY);
```
Comment (ishantiw) [inline framework/src/application.ts:235]: we can use `this.channel`

File: framework/src/application.ts [hunk 2]
```diff
@@ -126,19 +122,25 @@ export class Application {
...
+			appConfig: rootConfigs,
+			pluginConfigs: plugins,
+		});
 	}
 
 	public get networkIdentifier(): Buffer {
 		return this._node.networkIdentifier;
 	}
 
+	public get channel(): InMemoryChannel {
```
Comment (ishantiw) [inline framework/src/application.ts:140]: before `app.run` if user accesses `app.channel` will it throw error? or should return be `InMemory | undefined`

File: framework/test/unit/controller/http/http_server.spec.ts [hunk 3]
```diff
@@ -15,33 +15,32 @@ import * as HTTP from 'http';
 import { HTTPServer } from '../../../../src/controller/http/http_server';
 
 describe('HTTPServer', () => {
+	const logger = {
```
Comment (ishantiw) [inline framework/test/unit/controller/http/http_server.spec.ts:18]: can we use channelMock from testing here?

**Review by ishantiw — COMMENTED** [review]
LGTM, just some minor comments

**Review by Incede — APPROVED** [review]

**Review by ishantiw — APPROVED** [review]

**Pull request merged**